scalanet/src/io/iohk/scalanet/codec/FramingCodec.scala

```diff
@@ -0,0 +1,79 @@
+package io.iohk.scalanet.codec
+
+import java.nio.ByteBuffer
+
+import io.iohk.decco.{BufferInstantiator, Codec}
+import io.iohk.decco.Codec.{DecodeResult, Failure}
+import io.iohk.scalanet.codec.FramingCodec.State._
+import io.iohk.scalanet.codec.FramingCodec._
+
+import scala.collection.mutable
+
+class FramingCodec[T](messageCodec: Codec[T]) extends StreamCodec[T] {
+
+  private var state: State = LengthExpected
+  private var length: Int = 0
+  private var db: ByteBuffer = _
+  private val nlb = ByteBuffer.allocate(4)
+
+  override def streamDecode[B](source: B)(implicit bi: BufferInstantiator[B]): Seq[T] = this.synchronized {
```

Comment:
If I understand correctly, in the case of corrupted bytes where you receive a length but there aren't enough bytes to consume, the outcome is that you will get an empty list.

If there are enough bytes to satisfy the expected length but those bytes are corrupted, you just don't get `T` which ends up being another empty list.

Wouldn't it be worth to log a trace on these scenarios?

Reply:
All the bytes coming from the TCP stack will have been checksummed so the corruption should not happen.

What could happen is the connection could break somewhere during a read but that should be handled by the code managing the connection.

What I have noticed is that it would be very easy to send a rogue length parameter then try to fill the nodes memory with random bytes. I've added a maxBufferSize parameter to stop this (this was something supported by LengthFieldDecoder).

Reply:
I was not referring to corruption while sending bytes on the wire but corruption on purpose, thanks.